File: speeding_fine_tshirt/Graph/ScriptTemplate/LessEqual/CGLessEqual.js
/**
 * @file CGLessEqual.js
 * @author
 * @date 2021/8/15
 * @brief CGLessEqual.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGLessEqual extends BaseNode {
  constructor() {
    super();
  }

  getOutput() {
    return this.inputs[0]() <= this.inputs[1]();
  }
}

exports.CGLessEqual = CGLessEqual;

File: speeding_fine_tshirt/Graph/ScriptTemplate/LessEqual/CGLessEqual.lua
local CGLessEqual = CGLessEqual or {}
CGLessEqual.__index = CGLessEqual

function CGLessEqual.new()
    local self = setmetatable({}, CGLessEqual)
    self.inputs = {}
    return self
end

function CGLessEqual:setInput(index, func)
    self.inputs[index] = func
end

function CGLessEqual:getOutput(index)
    return self.inputs[0]() <= self.inputs[1]()
end

return CGLessEqual
